File: CS/EditManyToMany/PersistentClasses.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.Xpo;

namespace DataObjects {
    public class Department: XPObject {
        public Department(Session session)
            : base(session) { }

        private string _Name;
        public string Name {
            get {
                return _Name;
            }
            set {
                SetPropertyValue("Name", ref _Name, value);
            }
        }

        [Association("Locations-Departments")]
        public XPCollection<Location> Locations {
            get {
                return GetCollection<Location>("Locations");
            }
        }
    }

    public class Location: XPObject {
        public Location(Session session)
            : base(session) { }

        private string _Region;
        public string Region {
            get {
                return _Region;
            }
            set {
                SetPropertyValue("Region", ref _Region, value);
            }
        }

        [Association("Locations-Departments")]
        public XPCollection<Department> Departments {
            get {
                return GetCollection<Department>("Departments");
            }
        }
    }
}

File: VB/EditManyToMany/PersistentClasses.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.Xpo

Namespace DataObjects
	Public Class Department
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property

		<Association("Locations-Departments")> _
		Public ReadOnly Property Locations() As XPCollection(Of Location)
			Get
				Return GetCollection(Of Location)("Locations")
			End Get
		End Property
	End Class

	Public Class Location
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _Region As String
		Public Property Region() As String
			Get
				Return _Region
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Region", _Region, value)
			End Set
		End Property

		<Association("Locations-Departments")> _
		Public ReadOnly Property Departments() As XPCollection(Of Department)
			Get
				Return GetCollection(Of Department)("Departments")
			End Get
		End Property
	End Class
End Namespace
